test/test_smoke.py

```diff
@@ -134,3 +138,33 @@ def test_progress_smoke(tmp_path, build_fake_container, case: ProgressTestCase):
 
     assert case.needle in cast_text
     assert case.forbidden not in cast_text
+
+
+def test_smoke_force_data_dir(build_container):
+    """
+    Ensure that when a data dir is passed through `--force-data-dir` that only
+    distributions with repository files inside that directory are available.
+
+    Note that there's no 'negative' test case to this one, the default state of no
+    data directory is already tested by the other smoke tests.
+    """
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        repodir = pathlib.Path(tmpdir) / "repositories"
+        repodir.mkdir()
+
+        (repodir / "rhel-10.0.json").write_text(json.dumps({
+            "x86_64": [
+                {"name": "test", "baseurl": "test"},
+            ],
+        }))
+
+        output = subprocess.check_output(podman_run + [
+            "-v", f"{tmpdir!s}:/data",
+            build_container,
+            "--force-data-dir", "/data",
+            "list",
+        ], text=True)
+
+        # ensure we only have lines containing `rhel-10.0`
+        assert all("rhel-10.0" in line for line in output.splitlines())
```

Comment:
I would use `non-existing-os-1.0` just to make sure the code does not actually load our built-in OS repos. I guess the OS must exist, then could you perhaps use an architecture that is no longer supported in RHEL 10 and check for that combination? Assuming this is possible, if not, ignore my comment.

Reply:
That would require *also* passing in custom definitions.